lib/combineNodes.js

```diff
@@ -0,0 +1,290 @@
+            propertyNotShared(parent, child, 'skin') ||
+            propertyNotShared(parent, child, 'jointName')) {
+        return false;
+    }
+    if (preserveCameras && (parent.camera !== child.camera)) {
+        return false;
+    }
+
+    return true;
+}
+
+//Merges the child into the parent node, transforming the mesh data if needed.
+function mergeChildWithParent(gltf, parentId, childId, childIndex, preserveCameras) {
+    var nodes = gltf.nodes;
+    var parent = nodes[parentId];
+    var child = nodes[childId];
+    checkCombineWarning(child, 'node', childId);
+
+    var meshes = gltf.meshes;
+    //Duplicate and transform the child's mesh data if the node has a transformation, and add the mesh to the parent.
+    if (defined(child.meshes) && child.meshes.length > 0) {
+        var meshId = child.meshes[0];
+        var newMeshId = meshId;
+
+        var transform = child.matrix;
+        if (defined(transform)) {
+            transform = Matrix4.fromColumnMajorArray(transform);
+            if (!Matrix4.equals(transform, Matrix4.IDENTITY)) {
+                newMeshId += '_0';
+                var idCount = 0;
+                var meshKeys = Object.keys(meshes);
+                while (meshKeys.indexOf(newMeshId) !== -1) {
+                    idCount++;
+                    newMeshId = newMeshId.slice(0, -1) + idCount;
+                }
+
+                var mesh = gltf.meshes[meshId];
+                var newMesh = clone(mesh, true);
+                var primitives = newMesh.primitives;
+                if (defined(primitives)) {
+                    var primitivesLength = primitives.length;
+                    for (var i = 0; i < primitivesLength; i++) {
+                        var attributes = primitives[i].attributes;
+                        if (defined(attributes)) {
+                            transformAttribute(gltf, meshId, transform, 'POSITION', attributes, gltf.accessors, gltf.buffers, gltf.bufferViews);
+                            transformAttribute(gltf, meshId, transform, 'NORMAL', attributes, gltf.accessors, gltf.buffers, gltf.bufferViews);
+                        }
+                    }
+                }
+                meshes[newMeshId] = newMesh;
+            }
+        }
+        if (defined(parent.meshes)) {
+            parent.meshes.push(newMeshId);
+        } else {
+            parent.meshes = [newMeshId];
+        }
+    }
+
+    //If we are preseving cameras, the two nodes should already have the same camera.
+    //If not, remove the property if the two nodes do not have the same camera.
+    if (!preserveCameras && propertyNotShared(parent, child, 'camera')) {
+        delete parent.camera;
+    }
+    delete parent.name;
+
+    delete nodes[childId];
+
+    return parent;
+}
+
+//Transforms the attribute data and creates a new accessor, buffer view, and buffer to access the data.

```

Comment:
This relies on buffers, etc. getting combined later in the pipeline.  Is everything in the right order in gltf-pipeline.js?  I just want to confirm that there aren't cases were we leave the model less efficient than before it went through the pipeline.